### PR TITLE
[TR-106][TR-105] Added commands voting_deploy, payout_deploy and payout_deposit

### PR DESCRIPTION
**Pull request by voith**
- Fixed all incompatible changes caused by https://github.com/TokenMarketNet/smart-contracts/pull/150
- Added bytecode linking code(stolen from populus)
- Added tests for newly added commands

### CODE REVIEW

File: sto/ethereum/utils.py [hunk 1]
```diff
@@ -103,9 +110,12 @@ def get_constructor_arguments(contract: Contract, args: Optional[list]=None, kwa
     """
 
     # return contract._encode_constructor_data(args=args, kwargs=kwargs)
-
     constructor_abi = get_constructor_abi(contract.abi)
 
+    # constructor_abi can be none in case of libraries
+    if constructor_abi is None:
```
Comment (voith) [inline sto/ethereum/utils.py]: bug fix: `constructor_abi` can be None for libraries like `MathLib`

File: sto/ethereum/linking.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
+import re
```
Comment (voith) [inline sto/ethereum/linking.py]: [Can be ignored] lifted from populus, needed to link bytecode for contracts dependent on libraries.

**Comment by miohtama** [comment]
Can you also add relevant documentation. E.g. instruction pages how to do dividend payouts?

**Comment by voith** [comment]
@miohtama Adding that in a separate PR. currently working on it.

File: sto/cli/main.py [hunk 3]
```diff
@@ -605,6 +611,122 @@ def kyc_manage(config: BoardCommmadConfiguration, whitelist_address):
...
+        'name': to_bytes(text=payout_name),
+        'URI': to_bytes(text=uri),
+        '_type': type,
+        '_hash': integer_hash(type),
+        '_options': [to_bytes(i) for i in options]
+    }
+    deploy_contract(config, contract_name='PayoutContract', constructor_args=args)
+
+
+@cli.command(name="payout-deposit")
```
Comment (villesundell) [inline sto/cli/main.py]: Originally I was thinking that this command could take care of the approve() as well, but I guess it's cleaner this way :slightly_smiling_face:

**Comment by villesundell** [comment]
Only one comment, otherwise looks good to me :+1:

**Pull request merged**